xds/src/test/java/io/grpc/xds/GrpcXdsClientImplDataTest.java

```diff
@@ -2342,6 +2344,113 @@ public void parseCluster_validateEdsSourceConfig() throws ResourceInvalidExcepti
         LoadBalancerRegistry.getDefaultRegistry());
   }
 
+  @Test
+  public void processCluster_parsesMetadata()
+      throws ResourceInvalidException, InvalidProtocolBufferException {
+    ClusterMetadataRegistry testRegistry = ClusterMetadataRegistry.getInstance();
+    testRegistry.registerParser(new ClusterMetadataRegistry.ClusterMetadataValueParser() {
+      @Override
+      public String getTypeUrl() {
+        return "type.googleapis.com/test.Type";
+      }
+
+      @Override
+      public Object parse(Any value) {
+        return "testValue";
+      }
+    });
+
+    Any typedFilterMetadata = Any.newBuilder()
+        .setTypeUrl("type.googleapis.com/test.Type")
+        .setValue(ByteString.copyFromUtf8("test"))
```

Comment:
The test is not checking that this value is actually used. I suggest doing some transform in the parser, like adding a suffix. That way we know the full flow of data operates. It'd probably also be good to add an assertion to `parse(Any)` so we know it is only called when we expect.

Reply:
Though I do not completely understand 2nd part of this comment, I have a commit and I'm assuming I understood it correctly.
> It'd probably also be good to add an assertion to parse(Any) so we know it is only called when we expect.